src/index.ts

```diff
@@ -130,12 +130,31 @@ class ServerlessCustomDomain {
     }
 
     /**
-     * Lifecycle function to add domain info to the CloudFormation stack's Outputs
+     * Lifecycle function to createDomain before deploy and add domain info to the CloudFormation stack's Outputs
      */
-    public async updateCloudFormationOutputs(): Promise<void> {
+    public async createOrGetDomainForCfOutputs(): Promise<void> {
+        const autoDomain = this.serverless.service.custom.customDomain.autoDomain;
+        if (autoDomain === true) {
+            this.serverless.cli.log("Creating domain name before deploy.");
+            await this.createDomains();
+        }
 
         await this.getDomainInfo();
 
+        if (autoDomain === true) {
+            const atLeastOneDoesNotExist = this.domains.some((domain) => !domain.domainInfo);
+            if (atLeastOneDoesNotExist === true) {
+                const waitFor =
+                    parseInt(this.serverless.service.custom.customDomain.autoDomainWaitFor, 10)
+                    || 120;
+                this.serverless.cli.log(`Waiting ${waitFor} seconds before starting deployment
+                    because first time creating domain`);
+
+                await sleep(waitFor);
```

Comment:
One thing I don't like here is waiting for the entire period. What do you think about polling every few seconds until the time elapses? Or is there a throttling concern for this API that I'm forgetting about?

Reply:
agreed - this should only happen for the first deployment, but I did almost implement that before raising the PR, but wanted to get feedback first in case it could be avoided.

that being said, I'll add this now :) thanks!

Reply:
@jconstance-amplify I added a polling interval of 3 seconds (didn't seem worth making configurable).  Let me know what you think :)